packages/tools/devtools/devtools-core/src/data-visualization/VisualTree.ts

```diff
@@ -74,6 +74,11 @@ export interface VisualNodeBase {
 	 * Inside of the porperty is an array of possible EditTypes to inform devtools-view to only show the corresponding edit options for the types allowed
 	 */
 	editProps?: { editTypes?: EditType[] };
+
+	/**
+	 * Schema data of the SharedTree
+	 */
+	sharedTreeSchemaData?: string | Record<string | number, string>;
```

Comment:
Leaf node: Render as `boolean | string`

Non-leaf node: Pass Record to the `TreeHeader` and visualize in the custom tooltip.